### ISSUE TITLE
clustering coefficient support

### ISSUE THREAD

**Issue by kennylbj** [graphology/graphology#408]
Is there a way to calculate the clustering coefficient for the graph and the node? Or any plan to support this? thanks.

**Comment by Yomguithereal** [comment]
Hello @kennylbj, I will need more details/context for your request. What do you call clustering coefficient here? Are we speaking about this definition (which is not the only one): https://networkx.org/documentation/stable/reference/algorithms/generated/networkx.algorithms.cluster.clustering.html#networkx.algorithms.cluster.clustering ?

**Comment by kennylbj** [comment]
@Yomguithereal Yes, that's exactly what I requested, Same definition can be found in [wikipeida](https://en.wikipedia.org/wiki/Clustering_coefficient#:~:text=In%20graph%20theory%2C%20a%20clustering,graph%20tend%20to%20cluster%20together.).

**Comment by Yomguithereal** [comment]
Ok. Let me figure out if I need to enumerate actual triangles to compute the metric (in which case I must refactor my triangle enumerating routine using the Chiba-Nishizeki algorithm beforehand) or not. If I don't need that, it should be straightforward to implement (for unweighted graphs, at least).

**Comment by javieralaminos** [comment]
Hello,
Do we have any news on this issue? Or there is a way to calculate the clustering coefficient of a graph?
Thanks!

**Comment by Yomguithereal** [comment]
No news sorry. Did not have time nor incentive to work on this lately.

**Comment by javieralaminos** [comment]
Well, thats bad news :(
Could you point me in a direction in how to calculate this? We need to include this calculation in our graph.

**Comment by Yomguithereal** [comment]
networkx has some code to do this here: https://networkx.org/documentation/stable/reference/algorithms/generated/networkx.algorithms.cluster.clustering.html#networkx.algorithms.cluster.clustering if you want the metric for nodes. You could replicate it using graphology but this is not a simple thing to do.

Then if you "just" need some metric related to the importance of a node, maybe another of the centrality metrics already implemented could also fit the bill? The clustering coefficient is quite niche and I have never seen it used in the wild, but again I am coming from visual network analysis and I don't know all scientific disciplines that might absolutely need the clustering coefficient for very good reasons beyond my knowledge.

**Comment by gnlow** [comment]
Maybe this would work?
Not sure I understand the concept correctly.
```js
import { subgraph } from "graphology-operators"

const clustering =
(graph, node) => {
    const nbs = graph.neighbors(node)

    const t = subgraph(graph, nbs).edges().length
    const k = nbs.length

    return 2*t/(k*(k-1))
}
```

**Comment by Yomguithereal** [comment]
I believe this is the difference between local & global clustering coefficient?
